src/index.ts

```diff
@@ -0,0 +1,60 @@
+import net from "net";
+import {BeaconState, Peer} from "./types";
+import * as msgs from "./messages";
+
+interface HobbitsCtx {
+  port: number,
+  networkId: number,
+  chainId: number,
+  peers: Peer[],
+  staticPeers: Peer[],
+  state: BeaconState
+}
+
+class HobbitsP2PNetwork {
+  private port: number;
+  private networkId: number;
+  private chainId: number;
+  private peers: Peer[];
+  private staticPeers: Peer[];
+  private state: BeaconState;
+
+  constructor(ctx: HobbitsCtx) {
+    this.port = ctx.port;
+    this.networkId = ctx.networkId;
+    this.chainId = ctx.chainId;
+    this.peers = ctx.peers;
+    this.state = ctx.state;
+    this.staticPeers = ctx.staticPeers;
+  }
+
+  private connectStaticPeers = () => {
+    this.staticPeers.map((peer: Peer) => {
+      this.connect(peer);
+    })
+  };
+
+  private connect = (peer: Peer) => {
+    // TODO connect to the peer
+
+    this.sendHello()
+  };
+
+  private sendHello = () => {
+    const msg: msgs.Hello = {
+      network_id: this.networkId,
+      chain_id: this.chainId,
+      latest_finalized_root: this.state.latestFinalizedRoot,
+      latest_finalized_epoch: this.state.latestFinalizedEpoch,
+      best_root: this.state.latestFinalizedRoot,
+      best_slot: this.state.latestFinalizedSlot
+    }
+  };
+
+  private startServer = () => {
```

Comment:
isn't this suppose to be public, as well as connect and connect static peers?

It would be useful if this would return promise because: 
```
server.on('error', (e) => {
//port taken or something
})
```

Reply:
Good point thank you :)